app/lib/task/backend.dart

```diff
@@ -900,6 +905,45 @@ class TaskBackend {
     }
   }
 
+  /// Get status information for a package being analyzed.
+  Future<PackageStatus> packageStatus(String package) async {
+    final status = await cache.taskPackageStatus(package).get(() async {
+      final key = PackageState.createKey(_db, runtimeVersion, package);
+      final state = await dbService.lookupOrNull<PackageState>(key);
+
+      return PackageStatus(
+        package: package,
+        versions: (state?.versions ?? {}).entries.map((e) {
+          final v = e.key;
+          final s = e.value;
+
+          var status = TaskPackageVersionStatus.completed;
+          if (s.attempts > 0 && s.attempts < taskRetryLimit) {
+            // attempts > 0 && attempts < taskRetryLimit  ==> pending
+            status = TaskPackageVersionStatus.pending;
+          } else if (s.scheduled
+              .add(Duration(days: 31))
+              .isBefore(clock.now())) {
+            // scheduled + 31 days < now                  ==> pending
+            status = TaskPackageVersionStatus.pending;
+          } else if (s.attempts == 0 && s.scheduled == initialTimestamp) {
+            // attempts == 0 && scheduled == init         ==> pending
+            status = TaskPackageVersionStatus.pending;
+          } else if (s.attempts >= taskRetryLimit) {
+            // attempts >= taskRetryLimit                 ==> failed
+            status = TaskPackageVersionStatus.failed;
+          } else {
+            // attempts == 0                              ==> completed
```

Comment:
Btw. I find this a bit confusing that we are using the attempt count as a status indicator. It may be worth to add a separate flag that gets updated?

Reply:
Perhaps, but if we want to change the state, I think we should do this later.

Reply:
Now I remember we I didn't do this... obviously, after I started to try.

You cannot store the state when the state changes under you. A task is pending, if it haven't been analyzed in a certain amount of time. So storing field that says `state = "completed"` will be wrong a some point in time in the future.